core-rust/core-api-server/src/core_api/handlers/transaction_preview.rs

```diff
@@ -93,13 +99,33 @@ fn extract_preview_request(
 fn to_api_response(
     context: &MappingContext,
     result: ProcessedPreviewResult,
+    should_include_toolkit_receipt: bool,
 ) -> Result<models::TransactionPreviewResponse, ResponseError<()>> {
     let engine_receipt = result.receipt;
     let versioned_engine_receipt = engine_receipt.clone().into_versioned();
 
-    // This is interpreted by the toolkit in the wallet
+    // This is interpreted by the toolkit in the wallet. This will be removed with the release of
+    // the cuttlefish protocol update.
     let encoded_receipt = to_hex(scrypto_encode(&versioned_engine_receipt).unwrap());
 
+    // Produce a toolkit transaction receipt for the transaction preview if it was requested in the
+    // request opt-ins.
+    let toolkit_receipt = if should_include_toolkit_receipt {
+        Some(
+            RuntimeToolkitTransactionReceipt::try_from(versioned_engine_receipt.clone())
```

Comment:
Very minor (and I don't expect we need to change this) - given that `RuntimeToolkitTransactionReceipt` is a temporary model, it's slightly annoying that its creation  requires an owned `versioned_engine_receipt` rather than a reference. (In fact, it could even just store references into the receipt).

Reply:
Very true, I think that there's a world where we move towards that direction of making some of the data in the runtime toolkit receipt be references. I think that having everything be owned by the runtime receipt was easier, but I definitely agree that it is unnecessary.